lib/widgets/clipboard.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// Copies [data] to the clipboard and shows a popup on success.
+///
+/// Must have a [Scaffold] ancestor.
+///
+/// On newer Android the popup is defined and shown by the platform. On older
+/// Android and on iOS, shows a [Snackbar] with [successText].
+///
+/// In English, the [successText] should be short, should start with a capital
+/// letter, and should have no ending punctuation: "{noun} copied".
+void copyWithPopup({
+  required BuildContext context,
+  required ClipboardData data,
+  required Text successText // TODO(i18n)
```

Comment:
```suggestion
  required Widget successContent,
```

In general, APIs that take a widget should avoid caring about what type of widget it is; the type should be just Widget, unless the code is actually going to make use of features of a more specific type.

(And then this todo-i18n comment can go at the call site.)